crates/fraud_proofs/global_merkle_root/storage/src/update.rs

```diff
@@ -305,6 +307,275 @@ impl TransactionOutputs for Transaction {
     }
 }
 
-// TODO(#2582): Add tests (https://github.com/FuelLabs/fuel-core/issues/2582)
-#[test]
-fn dummy() {}
+#[cfg(test)]
+#[allow(non_snake_case)]
+mod tests {
+    use super::*;
+
+    use fuel_core_storage::{
+        structured_storage::test::InMemoryStorage,
+        transactional::{
+            ReadTransaction,
+            WriteTransaction,
+        },
+        StorageAsRef,
+    };
+    use fuel_core_types::fuel_tx::{
+        Bytes32,
+        ContractId,
+        TxId,
+    };
+
+    use rand::{
+        rngs::StdRng,
+        Rng,
+        SeedableRng,
+    };
+
+    #[test]
+    /// When encountering a transaction with a coin output,
+    /// `process_output` should ensure this coin is
+    /// populated in the `Coins` table.
+    fn process_output__should_insert_coin() {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut storage_update_tx =
+            storage_tx.construct_update_merkleized_tables_transaction();
+
+        let tx_pointer = random_tx_pointer(&mut rng);
+        let utxo_id = random_utxo_id(&mut rng);
+        let inputs = vec![];
+
+        let output_amount = rng.gen();
+        let output_address = random_address(&mut rng);
+        let output = Output::Coin {
+            to: output_address,
+            amount: output_amount,
+            asset_id: AssetId::zeroed(),
+        };
+
+        // When
+        storage_update_tx
+            .process_output(tx_pointer, utxo_id, &inputs, &output)
+            .unwrap();
+
+        storage_tx.commit().unwrap();
+
+        let inserted_coin = storage
+            .read_transaction()
+            .storage_as_ref::<Coins>()
+            .get(&utxo_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(*inserted_coin.amount(), output_amount);
+        assert_eq!(*inserted_coin.owner(), output_address);
+    }
+
+    #[test]
+    /// When encountering a transaction with a contract created output,
+    /// `process_output` should ensure an appropriate contract UTxO is
+    /// populated in the `ContractCreated` table.
```

Comment:
On the contrary to what is said in the comment, the test checks the `ContractsLatestUtxo` table - is this intended?

Reply:
Nice observation. Yeah this is just my typo. I was probably thinking about the output type while writing the wrong table name 😅. I'll create a tiny follow-up PR to fix it.

Reply:
Addressed in this PR https://github.com/FuelLabs/fuel-core/pull/2703